jdaviz/configs/default/aida.py

```diff
@@ -0,0 +1,187 @@
+import numpy as np
+import astropy.units as u
+from astropy.coordinates import SkyCoord, Angle
+from astropy.wcs import WCS
+from gwcs.wcs import WCS as GWCS
+
+from jdaviz.utils import get_top_layer_index
+
+
+class AID:
+    """
+    Common API methods for image viewers in astronomy, called
+    the Astro Image Display API (AIDA)[1]_.
+
+    References
+    ----------
+    .. [1] https://github.com/astropy/astro-image-display-api/
+
+    """
+
+    def __init__(self, viewer):
+        self.viewer = viewer
+        self.app = viewer.jdaviz_app
+
+    def _get_image_glue_data(self, image_label):
+        if image_label is None:
+            i_top = get_top_layer_index(self.viewer)
+            image = self.viewer.layers[i_top].layer
+
+        else:
+            for lyr in self.viewer.layers:
+                image = lyr.layer
+                if image.label == image_label:
+                    break
+            else:
+                raise ValueError(f"No data with data_label `{image_label}` found in viewer.")
+
+        return image, image.label
+
+    def set_viewport(self, center=None, fov=None, image_label=None, **kwargs):
```

Comment:
should everything in here be wrapped in a `with delay_callback(self.viewer.state, 'x_min', 'x_max', 'y_min', 'y_max', 'zoom_center_x', 'zoom_center_y', 'zoom_radius')` to avoid multiple callbacks as center and zoom are set separately?

Reply:
Good idea.